truffle/src/main/java/org/jruby/truffle/nodes/core/RegexpNodes.java

```diff
@@ -126,6 +126,13 @@ public Object match(RubyRegexp regexp, RubyString string) {
             return regexp.matchCommon(string, true, false) != getContext().getCoreLibrary().getNilObject();
         }
 
+        @Specialization
+        public Object match(RubyRegexp regexp, RubySymbol symbol) {
+            notDesignedForCompilation();
+
+            return regexp.matchCommon(symbol.toRubyString(), true, false) != getContext().getCoreLibrary().getNilObject();
```

Comment:
This refactoring doesn't look straightforward because of the way it is implemented right now. The MatchData uses RubyString as source.